query/executor.go

```diff
@@ -289,6 +292,98 @@ func (e *Executor) WithLogger(log *zap.Logger) {
 	e.TaskManager.Logger = e.Logger
 }
 
+func initQueryLogWriter(log *zap.Logger, e *Executor, path string) (*os.File, error) {
+	logFile, err := os.OpenFile(path, os.O_CREATE|os.O_RDWR|os.O_APPEND, 0644)
+	if err != nil {
+		e.Logger.Error("failed to open log file", zap.Error(err))
+		return nil, err
+	}
+
+	existingCore := log.Core()
+
+	encoderConfig := zap.NewProductionEncoderConfig()
+
+	fileCore := zapcore.NewCore(
+		zapcore.NewJSONEncoder(encoderConfig),
+		zapcore.Lock(logFile),
+		zapcore.InfoLevel,
+	)
+
+	newCore := zapcore.NewTee(existingCore, fileCore)
+	e.Logger = zap.New(newCore)
+	e.TaskManager.Logger = e.Logger
+
+	return logFile, nil
+}
+
+func closeQueryLogWriter(fd uintptr, e *Executor) {
+	// Uses the file descriptor pointer since we can remove or rename files
+	if err := syscall.Fsync(int(fd)); err != nil {
+		e.Logger.Error("failed to sync log file", zap.Error(err))
+	}
+	if err := syscall.Close(int(fd)); err != nil {
+		e.Logger.Error("failed to close log file", zap.Error(err))
+		return
+	}
+}
+
+func (e *Executor) WithLogWriter(log *zap.Logger, path string) {
+	var file *os.File
```

Comment:
Should the `closeQueryLogWriter` be in a `defer` call?  There's only one open at a time.